src/openms/source/FORMAT/MascotGenericFile.cpp

```diff
@@ -337,34 +337,66 @@ namespace OpenMS
       os << "BEGIN IONS\n";
       if (!store_compact_)
       {
-        os << "TITLE=" << precisionWrapper(mz) << "_" << precisionWrapper(rt)
-           << "_" << spec.getNativeID() << "_" << filename << "\n";
+        // if a TITLE is available, it was (most likely) parsed from an MGF file
+        // in that case, a nativeID was constructed from the TITLE and the spectrum index
+        if (spec.metaValueExists("TITLE"))
+        {
+          if (String(spec.getMetaValue("TITLE")).hasSubstring("_index="))
```

Comment:
is `_index` canonical? Could it be something else?

Reply:
`index=spectrum_number` is the native ID, defined by the mzTab and mzIdentML spectrum Id specification for MGFs. I added this to make sure all titles are unique, even if the loaded MGF had non-unique titles. Also this way the nativeIDs in ID files can somewhat be mapped back to the titles without relying purely on the order of spectra. The separator `_`was chosen arbitrarily and could be something else.

Here I just test whether the TITLE was already generated with the `else` case with `TITLE + "_" + nativeID`, meaning the loaded MGF was probably written out by OpenMS previously and already has the index. This is to prevent adding the index multiple times when processing MGFs with multiple OpenMS tools.

The nativeID is generated while reading in an MGF, so potentially it can be changed before writing the data out again, but usually OpenMS tools are not supposed overwrite nativeIDs of spectra.

Another option would be to make the new TITLE while reading in the file with `TITLE = TITLE + "_" + nativeID` and then writing out only the TITLE meta value. That way OpenMS tools or a pyOpenMS script could define their own titles freely by overwriting this, if necessary.